### PR TITLE
Microservice dev: LLM & vector DB for video RAG

### PR DESCRIPTION
**Pull request by shawnxhong**
## Description

vector DB microservice (FastAPI wrapped)
LLM inference microservice (Ray serve + LLama2)
streamlit pipeline updated to fetch results from restful api instead of from langchain lib

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds new functionality)

## Dependencies

see requirements.txt

## Tests

### CODE REVIEW

File: VideoRAGQnADistributed/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Visual RAG
+
+## Introduction
```
Comment (ttrigui) [inline VideoRAGQnADistributed/README.md:3]: Can you please update README with the one under VideoRAGQnA?

Comment (shawnxhong) [reply VideoRAGQnADistributed/README.md:3]: > Can you please update README with the one under VideoRAGQnA?

Hi I have updated the readme please kindly check

File: VideoRAGQnADistributed/embedding/extract_store_frames.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            
+            frame_count += 1
+            
+            if frame_count % mod == 0:
+                timestamp = cap.get(cv2.CAP_PROP_POS_MSEC) / 1000  # Convert milliseconds to seconds
+                frame_path = os.path.join(image_output_dir, f"{video}_{frame_count}.jpg")
+                
+                cv2.imwrite(frame_path, frame)  # Save the frame as an image
+                
+                metadata[frame_count] = {"timestamp": timestamp, "frame_path": frame_path}
```
Comment (ttrigui) [inline VideoRAGQnADistributed/embedding/extract_store_frames.py:51]: Metadata is missing, is there a reason for that?

**Review by ttrigui — COMMENTED** [review]
Can you please update the code with code under VideoRAGQnA

**Comment by gowthamhosamane** [comment]
At high level, changes look fine. Can be merged. @ttrigui

**Comment by ttrigui** [comment]
@shawnxhong can you please update the code from VideoRAGQnA folder?

**Comment by shawnxhong** [comment]
> @shawnxhong can you please update the code from VideoRAGQnA folder?

will make new PR.

**Pull request closed without merging**